apps/avatax/src/modules/avatax/calculate-taxes/avatax-calculate-taxes-payload-lines-transformer.ts

```diff
@@ -5,6 +5,74 @@ import { AvataxTaxCodeMatches } from "../tax-code/avatax-tax-code-match-reposito
 import { AvataxCalculateTaxesTaxCodeMatcher } from "./avatax-calculate-taxes-tax-code-matcher";
 import { avataxShippingLine } from "./avatax-shipping-line";
 
+type TaxBaseLine = TaxBaseFragment["lines"][0];
+
+const getUndiscountedTotalPrice = (line: TaxBaseLine) => {
+  if (line.sourceLine.__typename === "CheckoutLine") {
+    return line.sourceLine.undiscountedTotalPrice.amount;
+  }
+
+  return line.sourceLine.undiscountedTotalPrice.net.amount;
+};
+
+export const checkIfIsDiscountedLine = (isDiscounted: boolean, line: TaxBaseLine) => {
+  if (isDiscounted) {
+    const undiscountedTotalPrice = getUndiscountedTotalPrice(line);
+    const totalPrice = line.totalPrice.amount;
+
+    return totalPrice !== undiscountedTotalPrice;
+  }
+
+  return false;
+};
+
+/**
+ * If entire checkout discount then lines in webhook subscription are not discounted
+ * If one item is discounted then the line has discounted totalPrice
+ */
+export const checkDiscounts = (taxBase: TaxBaseFragment, isEntireSourceDiscounted: boolean) => {
```

Comment:
I'm against exporting such functions. If you were using classes, it will be likely a private class methods (which you should not test).

These methods afaik are exported only to test them. Its a code smell. Also you create a namespace bloating - we have now a lot of context-less functions that are not connected to any specific operation (do I check saleor model, avatax model etc)

Also, these methods are never called directly. AvataxCalculateTaxesPayloadLinesTransformer is called, so this is the public API.

We should test `AvataxCalculateTaxesPayloadLinesTransformer` methods that will verify if its working. "unit" doesn't mean "smallest possible function" but rather business unit that has complete behavior

Reply:
Okay, I'll just remove the test for these utils functions and keep the test for `[specific item voucher]`

Reply:
Just ensure tests for te main class have dataset/input that is diverse enough that covers these conditions